scylla/src/transport/connection.rs

```diff
@@ -801,6 +801,7 @@ impl Connection {
     pub(crate) async fn query_single_page(
         &self,
         query: impl Into<Query>,
+        page_size: PageSize,
     ) -> Result<QueryResult, QueryError> {
```

Comment:
Commit: `connection: pass page size through request API `
Could you explain here (and preferably in commit message too) why is there a need to have page_size argument for functions in Connection?
It was my understanding that page_size is attached to a statement (prepared or not) - in case of a function that I'm commenting on it would be in the `query: impl Into<Query>,`.

Reply:
It seems that I forgot to make `Connection`'s inner API follow the idea introduced for `Session`. Namely, `unpaged` method not taking `PageSize`, `single_page` taking `PageSize`, and both those methods based on top of a common method taking `Option<PageSize>`.

Reply:
I'll adjust it, thanks for pointing this out.